lib/vrl/stdlib/src/parse_ruby_hash.rs

```diff
@@ -0,0 +1,264 @@
+use nom::{
+    branch::alt,
+    bytes::complete::{escaped, tag, take_while},
+    character::complete::{alphanumeric1 as alphanumeric, char, one_of},
+    combinator::{cut, map, value},
+    error::{context, ContextError, FromExternalError, ParseError},
+    multi::separated_list0,
+    number::complete::double,
+    sequence::{preceded, separated_pair, terminated},
+    IResult,
+};
+use std::collections::BTreeMap;
+use std::num::ParseIntError;
+use vrl::prelude::*;
+use vrl::Value;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseRubyHash;
+
+impl Function for ParseRubyHash {
+    fn identifier(&self) -> &'static str {
+        "parse_ruby_hash"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "parse ruby hash",
+            source: r#"parse_ruby_hash("{ \"test\" => \"value\", \"testNum\" => 0.2, \"testObj\" => { \"testBool\" => true, \"testNull\" => nil } }")"#,
+            result: Ok(r#"
+                {
+                    "test": "value",
+                    "testNum": 0.2,
+                    "testObj": {
+                        "testBool": true,
+                        "testNull": null
+                    }
+                }
+            "#),
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        Ok(Box::new(ParseRubyHashFn { value }))
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::BYTES,
+            required: true,
+        }]
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseRubyHashFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ParseRubyHashFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+        let input = value.try_bytes_utf8_lossy()?;
+        parse(&input)
+    }
+
+    fn type_def(&self, _: &state::Compiler) -> TypeDef {
+        TypeDef::new().infallible().object::<(), Kind>(type_def())
```

Comment:
Parsing can fail, so this function has to be fallible.

```suggestion
        TypeDef::new().fallible().object::<(), Kind>(type_def())
```